src/rgw/rgw_iam_policy.cc

```diff
@@ -572,7 +572,7 @@ bool ParseState::do_string(CephContext* cct, const char* s, size_t l) {
   } else if ((w->id == TokenID::Action) ||
 	     (w->id == TokenID::NotAction)) {
     is_action = true;
-    if (*s == '*') {
+    if (s && *s == '*') {
```

Comment:
can "s" even be null?
ine lines 535, 549 and 554, we build a string/string_view based on "s" and "l".
if "s" could be null, we must verify that "l" is zero (otherwise we may crash)

Reply:
@vedanshbhartia ping